debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/rest/DebeziumMySqlConnectorResource.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql.rest;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.servlet.ServletContext;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.kafka.connect.health.ConnectClusterState;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.debezium.connector.mysql.Module;
+
+/**
+ * A JAX-RS Resource class defining endpoints of the Debezium MySQL Connect REST Extension
+ *
+ */
+@Path("/debezium/mysql")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class DebeziumMySqlConnectorResource {
+
+    private final Map<String, ?> config;
+
+    @javax.ws.rs.core.Context
+    private ServletContext context;
+
+    public DebeziumMySqlConnectorResource(ConnectClusterState clusterState, Map<String, ?> config) {
+        this.config = config;
+    }
+
+    @GET
+    @Path("/version")
+    public String getConnectorVersion() {
+        return Module.version();
+    }
+
+    @GET
+    @Path("/schema")
+    public JsonNode getConnectorSchema() {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Can this be passed on the constructor and instantiated on `*ConnectRestExtension`?

Reply:
A very good point, this is a copy and paste leftover from the first naive implementation.

Reply:
What do you think about this approach in general @mfvitale ? I could also just return the content of the file. The only reason why I did it that way was to ensure valid JSON is coming from that file.

Reply:
This can be an option but is still not easily testable. Passing it to the constructor you can easy tests the class passing also a mocked `ObjectMapper`. If this serves just for validating the json isn't it better to fail on the `register` method?

Reply:
Hummm, what logic exactly would you test except that the right file & content is being returned or that exception being thrown in case of failure? Especially when I compare it to the alternative solution (return just the file contents as String without further JSON parsing and "validation"). Using that ObjectMapper re-conversion is just some maybe unnecessary safety net.

Reply:
Should you catch `DataBindException` for "validation" error right? In the case of `IOException` you are just checking if able to read the file. In any case. do what you think is the best.

Reply:
@mfvitale Yes you are right DataBindException must be catched, but DataBindException is an IOException (subclass/extended class of IOException), so it is already catched with the IOException.